PBN RCT.Rmd

```diff
@@ -27,157 +27,210 @@ if(!require(openxlsx)){install.packages('openxlsx')}
-    na.rm = TRUE))
-#For each participant, calculate the percent of missing responses (there are 23 items total)
+#For each participant, add a new column that calculates the percent missing of baseline primary outcome items
 included_participants <- included_participants %>%
-   mutate(percent_missing = (1 - (total_responses_bprimary / 23)) * 100) 
+  mutate(percent_missing = rowMeans(
+    is.na(select(included_participants, b_mfq_1, b_mfq_2, b_mfq_3, b_mfq_4, b_mfq_5, b_mfq_6, b_mfq_7, b_mfq_8, b_mfq_9, b_mfq_10, b_mfq_11, b_mfq_12, b_mfq_13, b_edeq_1, b_edeq_2, b_edeq_3, b_edeq_4, b_edeq_5, b_edeq_6, b_edeq_7, b_edeq_8, b_edeq_9, b_edeq_10)), 
+    na.rm = TRUE)) %>%
+  mutate(percent_missing = percent_missing * 100)
+
 #Make a new variable that identifies participants with more than 5% missing data (those with more than 5% missing are TRUE)
 included_participants <- included_participants %>%
   mutate(more_than_5_percent_missing = percent_missing > 5)
 #Remove participants where more_than_5_percent_missing is TRUE
 included_participants <- included_participants %>%
   filter(!more_than_5_percent_missing)
 #In sum, there were originally 220 included_participants and now there are 218 (only had to drop two participants prior to imputation)
+###NEED TO CHECK### - I think included_participants can now be considered analysis-ready for imputation (since it dropped participants with more than 5% missing baseline data and includes all the variables that I think will be useful in imputation). There are a few extra variables that I've used along the way such as Participant_ID, treatment_group, percent_missing, and more_than_5_percent missing. Keep or drop these for imputation?
 ```
 
 ```{r imputation}
 set.seed(12345) #use the same seed number each time to make the code reproducible (the seed number is just the starting point used in the generation of a sequence of random numbers in the random forest)
 
-#Select the data needed for imputation (the outcome variables themselves and the variables that will go into the algorith WHICH I NEED TO ADD)
-variables_for_imputation <- included_participants %>%
-  select( b_mfq_1, b_mfq_2, b_mfq_3, b_mfq_4, b_mfq_5, b_mfq_6, b_mfq_7, b_mfq_8, b_mfq_9, b_mfq_10, b_mfq_11, b_mfq_12, b_mfq_13, b_edeq_1, b_edeq_2, b_edeq_3, b_edeq_4, b_edeq_5, b_edeq_6, b_edeq_7, b_edeq_8, b_edeq_9, b_edeq_10)
-
-imputed_data <- missForest(variables_for_imputation)
-imputed_data <- as.data.frame(imputed_data$ximp)
-imputed_data <- round(imputed_data) #Since the algorithm introduces decimals which is not appropriate for this data, round to the nearest whole number
+#x below will be the analysis-ready dataset with
+#imputed_data <- missForest(included_participants)
+#imputed_data <- as.data.frame(imputed_data$ximp)
```

Comment:
You'll also want to hang onto (or at least print out) the error estimates, to report in the manuscript